text/0000-associated-constants-on-ints.md

```diff
@@ -0,0 +1,269 @@
+
+Anticipating the situation, an [issue](https://github.com/rust-lang/rfcs/issues/1099) was filed in the RFCs repo regarding moving the contents of these modules into associated consts:
+
+> I think it's a minor enough breaking change to move the constants and deprecate the modules u8, u16, etc. Not so sure about removing these modules entirely, I'd appreciate that, but it'll break all the code use-ing them. [petrochenkov, Apr 29, 2015]
+
+Finally, so obvious was this solution that [the original RFC for associated items](https://github.com/nox/rust-rfcs/blob/master/text/0195-associated-items.md#expressiveness) used the numeric constants as the only motivating example for the feature of associated consts:
+
+> For example, today's Rust includes a variety of numeric traits, including Float, which must currently expose constants as static functions [...] Associated constants would allow the consts to live directly on the traits
+
+Despite the obvious intent, 1.0 came and went and there were plenty of other things to occupy everyone's attention. Now, two days shy of Rust's fourth anniversary, let's re-examine the situation. We propose to deprecate all of the aforementioned functions and constants in favor of associated constants defined on the appropriate types, and to additionally deprecate the `i8`, `i16`, `i32`, `i64`, `i128`, `isize`, `u8`, `u16`, `u32`, `u64`, `u128`, `usize`, `f32`, and `f64` modules in `std`. Advantages of this:
+
+1. Consistency with the rest of the language. As demonstrated by the above quotes, associated consts have been the natural way to express these concepts in Rust since before associated consts were even implemented; this approach satisfies the principle of least surprise.
+
+2. Documentation. On the front page of the [standard library API docs](https://doc.rust-lang.org/std/index.html), 14 of the 56 modules in the standard library (25%) are the aforementioned numeric modules whose only purpose is to namespace these constants. This number will continue to rise as new numeric primitives are added to Rust, as already seen with `i128` and `u128`. Although deprecated modules cannot be easily removed from std, they can be removed from the documentation, making the stdlib API docs less cluttered and easier to navigate.
+
+3. Beginner ease. For a beginner, finding two identical ways to achieve something immediately raises the question of "why", to which the answer here is ultimately uninteresting (and even then, the question of "which one to use" remains unanswered; neither current approach is idiomatic). As noted, deprecated items can be removed from the documentation, thereby decreasing the likelihood of head-scratching and incredulous sidelong glances from people new to Rust.
```

Comment:
I think the motivation doesn't sufficiently justify why the current constants are harmful. Meanwhile, I think there's plenty of harm caused by deprecation of constants that on initial inspection seem like they would be widely used. At minimum, I think there should be a crater run to see how much churn there would be.